aws-iot-softwarepackage/src/main/java/software/amazon/iot/softwarepackage/CreateHandler.java

```diff
@@ -0,0 +1,188 @@
+     * @param proxyClient the aws service client to make the call
+     * @return awsResponse update resource response
+     */
+    private UpdatePackageVersionResponse updateResourceForPackageVersion(
+            final UpdatePackageVersionRequest updatePackageVersionRequest,
+            final ProxyClient<IotClient> proxyClient) {
+        try {
+            final UpdatePackageVersionResponse updatePackageVersionResponse = proxyClient.injectCredentialsAndInvokeV2(
+                    updatePackageVersionRequest, proxyClient.client()::updatePackageVersion);
+            logger.log(String.format("%s [%s, %s] successfully updated.",
+                    ResourceModel.TYPE_NAME, updatePackageVersionRequest.packageName(), updatePackageVersionRequest.versionName()));
+            return updatePackageVersionResponse;
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToHandlerException(updatePackageVersionRequest.packageName(), OPERATION, e);
+        }
+    }
+
+    /**
+     * Implement client invocation of the create request through the proxyClient, which is already initialised with
+     * caller credentials, correct region and retry settings
+     *
+     * @param updatePackageRequest     the aws service request to update a resource
+     * @param proxyClient the aws service client to make the call
+     * @return awsResponse update resource response
+     */
+    private UpdatePackageResponse updateResourceForDefaultVersionName(
+            UpdatePackageRequest updatePackageRequest,
+            ProxyClient<IotClient> proxyClient) {
+        try {
+            final UpdatePackageResponse updatePackageResponse = proxyClient.injectCredentialsAndInvokeV2(
+                    updatePackageRequest, proxyClient.client()::updatePackage);
+            logger.log(String.format("%s [%s] has been successfully updated.",
+                    ResourceModel.TYPE_NAME, updatePackageRequest.packageName()));
+            return updatePackageResponse;
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToHandlerException(updatePackageRequest.packageName(), OPERATION, e);
+        }
+    }
+
+    private String generateName(final ResourceHandlerRequest<ResourceModel> request) {
+        final StringBuilder identifierPrefix = new StringBuilder();
+        identifierPrefix.append((request.getSystemTags() != null &&
+                MapUtils.isNotEmpty(request.getSystemTags())) ?
+                request.getSystemTags().get("aws:cloudformation:stack-name") + "-" : "");
+        identifierPrefix.append(request.getLogicalResourceIdentifier() == null ?
+                "SOFTWARE-PACKAGE" :
+                request.getLogicalResourceIdentifier());
+
+        return IdentifierUtils.generateResourceIdentifier(
+                identifierPrefix.toString(),
+                request.getClientRequestToken(),
+                MAX_PACKAGE_NAME_LENGTH);
+    }
```

Comment:
If I'm creating a cfn resurce with package name ''sp", and tag "tag", what will the final name look like